### DISCUSSION TITLE
Problem in preFetch information، when we change the parameter on the page

### DISCUSSION THREAD

**Discussion by abbasvaranloo** [quasarframework/quasar#11781] [CLI - SSR mode] [answered]
I use Quasar V2 for may app.

This is my route for getting information based on city parameter:
{
path: 'business/:city',
name: 'business',
component: () => mainComponent,
children: [
{
path: '',
name: 'all-business',
component: () => component,
},
]
}
prefetch works fine if I go that page via another pages, But if I change the route parameter or refresh the page, my app crashes, Axios seems to try several times to get information.

` preFetch({store, currentRoute, previousRoute, redirect, ssrContext}) {

    store.registerModule('business', businessStoreModule)

    return store.dispatch('business/fetchBusinessData', currentRoute)
  }`

and in setup :

`
const store = useStore()

    onMounted(() => {
      store.registerModule('business', businessStoreModule, {preserveState: true})
    })

    onUnmounted(() => {
      store.unregisterModule('business')
    })`

**Comment by rstoenescu** [comment, marked as answer]
Given your code, you are registering the module each time the preFetch() is run. Which can happen multiple times if user navigates back and forth to that route. Guard the registerModule call.

**Reply by abbasvaranloo** [reply]
Yes you are right, thanks for answer.
